libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/State/QuerySpec.hs

```diff
@@ -51,166 +66,185 @@ spec = do
 propComplete nes = do
-  withCommitteeInfo nes $ \comMembers _ comStateMembers noFilterResult -> do
-    -- if a credential appears in either Committee or CommitteeState, it should appear
-    -- in the result
-    Map.keysSet comMembers
-      `Set.union` Map.keysSet comStateMembers
-      `shouldBe` Map.keysSet (csCommittee noFilterResult)
+  withCommitteeInfo nes $
+    \(Committee comMembers _) (CommitteeState comStateMembers) nextComMembers noFilterResult -> do
+      -- if a credential appears in either Committee or CommitteeState, it should appear
+      -- in the result
+      Set.unions [Map.keysSet comMembers, nextComMembers, Map.keysSet comStateMembers]
+        `shouldBe` Map.keysSet (csCommittee noFilterResult)
 
 propNotAuthorized ::
   forall era.
   EraGov era =>
   NewEpochState era ->
   Expectation
 propNotAuthorized nes = do
-  withCommitteeInfo nes $ \_comMembers _ comStateMembers noFilterResult -> do
-    let notAuthorized =
-          Map.filter
-            ( \case
-                CommitteeMemberState MemberNotAuthorized _ _ _ -> True
-                _ -> False
-            )
-            (csCommittee noFilterResult)
-    -- if the member is NotAuthorized, it should not have an associated hot credential in the committeeState
-    Map.intersection comStateMembers notAuthorized `shouldBe` Map.empty
+  withCommitteeInfo nes $
+    \_ (CommitteeState comStateMembers) _ noFilterResult -> do
+      let notAuthorized =
+            Map.filter
+              ( \case
+                  CommitteeMemberState MemberNotAuthorized _ _ _ -> True
+                  _ -> False
+              )
+              (csCommittee noFilterResult)
+      -- if the member is NotAuthorized, it should not have an associated hot credential in the committeeState
+      Map.intersection comStateMembers notAuthorized `shouldBe` Map.empty
 
 propAuthorized ::
   forall era.
   EraGov era =>
   NewEpochState era ->
   Expectation
 propAuthorized nes = do
-  withCommitteeInfo nes $ \_comMembers _ comStateMembers noFilterResult -> do
-    let ckHk =
-          Map.foldMapWithKey
-            ( \ck cms ->
-                case cms of
-                  CommitteeMemberState (MemberAuthorized hk) _ _ _ -> [(ck, Just hk)]
-                  _ -> []
-            )
-            (csCommittee noFilterResult)
-    -- if the member is Authorized, it should appear in the commiteeState
-    Map.filter isJust comStateMembers `shouldBe` Map.fromList ckHk
+  withCommitteeInfo nes $
+    \_ (CommitteeState comStateMembers) _ noFilterResult -> do
+      let ckHk =
+            Map.foldMapWithKey
+              ( \ck cms ->
+                  case cms of
+                    CommitteeMemberState (MemberAuthorized hk) _ _ _ -> [(ck, Just hk)]
+                    _ -> []
+              )
+              (csCommittee noFilterResult)
+      -- if the member is Authorized, it should appear in the commiteeState
+      Map.filter isJust comStateMembers `shouldBe` Map.fromList ckHk
```

Comment:
Nice.  Looking at the documentation for this function, I realise that "Map.mapMaybe" could be even better, since I use the key only to create the entry: 

```
      let ckHk =
            Map.mapMaybe
              ( \case
                    CommitteeMemberState (MemberAuthorized hk) _ _ _ -> Just hk
                    _ -> Nothing
              )
              (csCommittee noFilterResult)
```

Reply:
Yeah, `mapMaybe` is definitely better. My in-github-pr Haskell coding isn't as good as I would like it to be :smile: